### PR TITLE
Add Operational Best Practices for FFIEC. Closes #418

### LINKED ISSUE #418: Add Operational Best Practices for FFIEC
**Is your feature request related to a problem? Please describe.**

https://docs.aws.amazon.com/config/latest/developerguide/operational-best-practices-for-ffiec.html

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by vkumbha**
### Checklist
- [x] Issue(s) linked

### CODE REVIEW

File: ffiec/d_1/d_1.sp [hunk 1]
```diff
@@ -0,0 +1,17 @@
+locals {
+  d_1_common_tags = merge(local.ffiec_common_tags, {
+    ffiec_domain = "1"
+  })
+}
+
+benchmark "d_1" {
+  title         = "Cyber Risk Management and Oversight (Domain 1)"
```
Comment (cbruno10) [inline ffiec/d_1/d_1.sp:8]: Can you please align the `=`'s here and other places?

File: ffiec/ffiec.sp [hunk 2]
```diff
@@ -0,0 +1,19 @@
+locals {
+  ffiec_common_tags = merge(local.aws_compliance_common_tags, {
+    ffiec = "true"
+    type  = "Benchmark"
+  })
+}
+
+benchmark "ffiec" {
+  title       = "Federal Financial Institutions Examination Council (FFIEC)"
+  description = "The Council is a formal interagency body empowered to prescribe uniform principles, standards, and report forms for the federal examination of financial institutions by the Board of Governors of the Federal Reserve System (FRB), the Federal Deposit Insurance Corporation (FDIC), the National Credit Union Administration (NCUA), the Office of the Comptroller of the Currency (OCC), and the Consumer Financial Protection Bureau (CFPB), and to make recommendations to promote uniformity in the supervision of financial institutions."
```
Comment (rajlearner17) [inline ffiec/ffiec.sp:10]: We should link the ffiec_overview.md here as below

  documentation = file("./ffiec/docs/ffiec_overview.md")

File: conformance_pack/backup.sp [hunk 3]
```diff
@@ -24,20 +25,32 @@ control "backup_plan_min_retention_35_days" {
...
     nist_csf = "true"
     soc_2    = "true"
   })
-}
\ No newline at end of file
+}
+
+control "backup_recovery_point_min_retention_35_days" {
+  title       = "Backup recovery points should not expire before retention period"
+  description = "Ensure if a recovery point expires no earlier than after the specified period. The rule is non_compliant if the recovery point has a retention point less than 35 days."
```
Comment (misraved) [inline conformance_pack/backup.sp:50]:
```suggestion
  description = "Ensure a recovery point expires no earlier than after the specified period. The rule is non-compliant if the recovery point has a retention point less than 35 days."
```

File: ffiec/d_1/d_1_rm.sp [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  children = [
+    benchmark.ffiec_d_1_rm_rm_b_1,
+    benchmark.ffiec_d_1_rm_ra_b_2
+  ]
+
+  tags = local.ffiec_d_1_common_tags
+}
+
+benchmark "ffiec_d_1_rm_rm_b_1" {
+  title       = "D1.RM.Rm.B.1"
```
Comment (misraved) [inline ffiec/d_1/d_1_rm.sp:13]: I think we should use the titles as per the official documentation. We could change them once - https://github.com/awsdocs/aws-config-developer-guide/issues/21 is resolved

File: ffiec/d_2/d_2_ma.sp [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    control.vpc_flow_logs_enabled,
+    control.s3_bucket_object_lock_enabled,
+    control.cloudtrail_s3_data_events_enabled,
+    control.cloudwatch_log_group_retention_period_365,
+    control.rds_db_instance_logging_enabled,
+    control.wafv2_web_acl_logging_enabled,
+    control.redshift_cluster_kms_enabled,
+    control.redshift_cluster_encryption_logging_enabled,
+    control.cloudtrail_trail_integrated_with_logs,
+    control.es_domain_logs_to_cloudwatch
```
Comment (misraved) [inline ffiec/d_2/d_2_ma.sp:30]: Please sort the controls

File: ffiec/d_2/d_2_ma.sp [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    control.cloudtrail_trail_integrated_with_logs,
+    control.cloudtrail_trail_enabled,
+    control.elb_application_classic_lb_logging_enabled,
+    control.cloudtrail_multi_region_trail_enabled,
+    control.s3_bucket_logging_enabled,
+    control.vpc_flow_logs_enabled,
+    control.cloudtrail_s3_data_events_enabled,
+    control.redshift_cluster_kms_enabled,
+    control.redshift_cluster_encryption_logging_enabled,
+    control.es_domain_logs_to_cloudwatch
```
Comment (misraved) [inline ffiec/d_2/d_2_ma.sp:52]: Please sort the controls.

File: ffiec/d_2/d_2_ti.sp [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  tags = local.ffiec_d_2_common_tags
+}
+
+benchmark "ffiec_d_2_ti_ti_b_1" {
+  title       = "D2.TI.Ti.B.1"
+  description = "The institution belongs or subscribes to a threat and vulnerability information-sharing source(s) that provides information on threats (e.g., FS-ISAC, US- CERT)."
+  children = [
+    control.securityhub_enabled,
+    control.guardduty_enabled
+
```
Comment (misraved) [inline ffiec/d_2/d_2_ti.sp:19]:
```suggestion
```

File: query/backup/backup_recovery_point_min_retention_35_days.sql [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+  recovery_point_arn as resource,
+  case
+    when (lifecycle -> 'DeleteAfterDays') is null then 'ok'
+    when (lifecycle -> 'DeleteAfterDays')::int >= 35 then 'ok'
+    else 'alarm'
+  end as status,
+  case
+    when (lifecycle -> 'DeleteAfterDays') is null then split_part(recovery_point_arn, ':', -1) || ' retention period set to never expire.'
+    when (lifecycle -> 'DeleteAfterDays')::int >= 35 then split_part(recovery_point_arn, ':', -1) || ' recovery point expires after retention period.'
+    else split_part(recovery_point_arn, ':', -1) || ' recovery point expires before retention period.'
```
Comment (misraved) [inline query/backup/backup_recovery_point_min_retention_35_days.sql:12]:
```suggestion
    else split_part(recovery_point_arn, ':', -1) || ' recovery point has a retention period of ' || (lifecycle -> 'DeleteAfterDays')::int || ' days.'
```

Comment (misraved) [reply query/backup/backup_recovery_point_min_retention_35_days.sql:12]: Could we update the query to use a single line for the reason ?

**Review by misraved — COMMENTED** [review]
The controls listed under benchmarks need to be sorted. I have provided comments in a few benchmarks, please verify this across all the other benchmarks.

**Comment by misraved** [comment]
Please make changes wherever required 👍

**Review by misraved — APPROVED** [review]

**Pull request merged**